File: modules/test_sampler1.lua
FunctionSampler1D = require('FunctionSampler1D')

function f(x)
	return math.sin(1/(x*x+0.05))
end

sampler = FunctionSampler1D.New()

sampler:Add(-4, f(-4))
sampler:Add( 0, f( 0))
sampler:Add( 4, f( 4))

while not sampler:IsDone() do
	local xvals = sampler:GetNext()
	for ix,x in ipairs(xvals) do
		local y = f(x)
		sampler:Add(x, y)
		print(x, y)
	end
end

File: modules/test_sampler1.py
import math
import FunctionSampler1D

sampler = FunctionSampler1D.new()

def f(x):
	return math.sin(1/(x*x+0.05))


sampler.add(-4, f(-4))
sampler.add(0, f(0))
sampler.add(4, f(4))

while not sampler.is_done():
	xvals = sampler.get_next()
	for x in xvals:
		y = f(x)
		sampler.add(x, y)

print('#', len(sampler))
for t in sampler:
	x, y, id = t;
	print(x, y)